project/Version.scala

```diff
@@ -30,4 +30,26 @@ object Version {
       Seq(file)
     }.taskValue
   )
+
+  lazy val generateVersionClass = Seq(
+    sourceGenerators in Compile += Def.task {
+      val file = (sourceManaged in Compile).value / "com" / "commercetools" / "sunrise" / "common" / "version" / "SunriseSolutionInfo.java"
+      IO.write(file,
+        s"""package com.commercetools.sunrise.common.version;
+          |
+          |import io.sphere.sdk.client.SolutionInfo;
+          |
+          |public class SunriseSolutionInfo extends SolutionInfo {
+          |
+          |
+          |    public SunriseSolutionInfo() {
+          |        setName("sunrise-java-shop-framework");
+          |        setVersion("${version.value}");
+          |    }
+          |
+          |}
+          |""".stripMargin)
+      Seq(file)
+    }.taskValue)
+
```

Comment:
Actually this plugin generate scala code, but i would agree the way i did it wouldn't let us now if some error happened during the build process thats why i will build a class BuildInfo in using the same way i generated SunriseVersionInfo.java, the only diff is that i'm gonna refernece this class in SunriseSolutionInfo, so if an error happens the code wont compile